ingestion-edge/ingestion_edge/publish.py

```diff
@@ -77,8 +78,8 @@ def flush(q: SQLiteAckQueue) -> Tuple[str, int]:
     """
     if q.size == 0:
         # early return for empty queue
-        return "", FLUSH_EMPTY_STATUS
-    done = 0
+        return response.raw(b"", HTTP_STATUS.NO_CONTENT)
+    result = {"done": 0, "pending": 0}
```

Comment:
Given that we make reference to these `done` and `pending` fields via string accessors several times, I'd like to  see them turned into string constants, or even better we can probably turn this into a class:

```
@dataclass 
Result:
  done: int = 0
  pending: int = 0

  def json(self):
    return json.dumps(self.__dict__)
```

Reply:
`.json` won't do because we should leave json encoding to `response.json`, making it `response.json(result.__dict__, <status>)` or `def as_dict(self): self.__dict__` or something.

seems like overkill, i'd rather go back to using two variables and a `dict` call.

Reply:
actually, i take that back, python's type system don't make either of those options notably safer than this.

Reply:
specifically note that the following works and passes flake8:
```from dataclasses import asdict, dataclass
import json


@dataclass
class Result:
    done: int = 0
    pending: int = 0


r = Result()
r.don = 1
print(json.dumps(asdict(r)))
```
and it prints `{"done": 0, "pending": 0}`

Reply:
long way of saying: i'll just use string constants